### DISCUSSION TITLE
INDI controller vs PID controller on Crazyflie

### DISCUSSION THREAD

**Discussion by YWprog** [bitcraze/discussions#1676] [Q&A] [answered]
Hello everyone, I'm wondering why Crazyflie's performance with the INDI controller was worse than that with the PID controller. 
I've been learning UAV controllers and found several papers and websites that mention that INDI is more accurate than the PID controller. I used Lighthouse to estimate Crazyflie's position, yet PID performed more stable than INDI under the same circumstances. Has anyone tried INDI before? Do you know if there is anything to be concerned about? Thanks in advanced.

**Comment by knmcguire** [comment]
Hi! 

Not many try out the INDI controller as compared to the Mellinger and PID controller, and internally we don't really test it for our releases either. Usually we see these controllers as more of the responsibility of those that contributed it, but we only test it if 'it flies at all'. 

There is an old ticket open about the INDI controller so I'll mark this discussion as part of it: https://github.com/bitcraze/crazyflie-firmware/issues/573

**Reply by YWprog** [reply]
Hi! 
Thanks for your quick reply and marked this discussion in the INDI main discussion. Though it was tested just to fly, wondering how those parameters were get in the initial, like attitude err proportional gain, derivative gains (p, q, r) and actuator model coefficient.

**Reply by gemenerik** [reply]
Just to clarify, are you asking about how the initial parameter values for the INDI controller (like the proportional and derivative gains or actuator model coefficients) were determined by its contributors? Let me know if that's the case or if you’re referring to something else!

**Reply by YWprog** [reply]
Yes, you got the point!

**Comment by gemenerik** [comment, marked as answer]
I think it might be best to read the original PR: https://github.com/bitcraze/crazyflie-firmware/pull/483
There is quite some discussion about parameter tuning there.
